src/main/groovy/org/checkerframework/gradle/plugin/CheckerFrameworkPlugin.groovy

```diff
@@ -101,43 +102,35 @@ final class CheckerFrameworkPlugin implements Plugin<Project> {
       }
     })
 
+    // Also apply the checker to all subprojects
+    if (userConfig.applyToSubprojects) {
+      project.subprojects { subproject -> apply(subproject) }
+    }
+
     project.gradle.projectsEvaluated {
       if (!applied) LOG.warn('No android or java plugins found, checker compiler options will not be applied.')
     }
   }
 
   private static configureProject(Project project, CheckerFrameworkExtension userConfig) {
-    JavaVersion javaVersion =
-        project.extensions.findByName('android')?.compileOptions?.sourceCompatibility ?:
-        project.property('sourceCompatibility')
 
-    // Check Java version.
-    def jdkVersion
-    if (javaVersion.java7) {
-      throw new IllegalStateException("The Checker Framework does not support Java 7.")
-    } else if (javaVersion.java8) {
-      jdkVersion = ANNOTATED_JDK_NAME_JDK8
-    } else {
-      // Use Java 8, even if the user requested a newer version of Java.
-      // Undo this hack when newer annotated JDKs are released by the Checker Framework team.
-      jdkVersion = ANNOTATED_JDK_NAME_JDK8
-    }
+    def jdkVersion = ANNOTATED_JDK_NAME_JDK8
```

Comment:
I found the distinction between jdkVersion here, and javaVersion on line 149, confusing.  When can/should they be different?  Is this one a name more than a version (that is suggested by the name of variable `ANNOTATED_JDK_NAME_JDK8`)?